docs/python.md

```diff
@@ -103,6 +103,19 @@ The `Input()` function takes these keyword arguments:
 
 Each parameter of the `predict()` method must be annotated with a type like `str`, `int`, `float`, `bool`, etc. See [Input and output types](#input-and-output-types) for the full list of supported types.
 
+Using the `Input` function is recommended because it provides better documentation and validation constraints to the users of your model, but it is not strictly required. You can also specify default values for your parameters using plain Python, or omit default assignment entirely:
+
+```py
+class Predictor(BasePredictor):
+    def predict(self, 
+        prompt: str = "default prompt", # this is valid
+        iterations: int                 # also valid
+    ) -> str:
+        for x in (iterations or 1:)
```

Comment:
👍

`(iterations or 1:)` [sic -- should be `):`] is also not really Pythonic. Python prefers clear, explicit things rather than clever one liners. It also wouldn't work in the case where it is `0`, because that is falsy.

Reply:
Yeah I knew this was a bad example! Couldn't think of a great way to include a parameter that doesn't have a default value, but still do something sensible with it. Maybe the body of the function should just be `# ...` to avoid distraction. The import bit is really just illustrating the way params can be written.